code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_core.py

```diff
@@ -0,0 +1,290 @@
+        """
+        Request the status of the CS by sending a request to the request channel.
+        """
+        log.debug("Checking CS status")
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.GET_CS_STATUS)
+        self.__chan_str_reqs.send_data(request)
+        self.__wait_cs_reqst = True
+
+    def __fetch_new_exp(self) -> None:
+        """AI is creating summary for fetch_new_exp
+
+        Raises:
+            ValueError: [description]
+
+        Returns:
+            Tuple[CyclerDataExperimentC, CyclerDataBatteryC, CyclerDataProfileC]: [description]
+        """
+        log.debug("Checking for new experiments")
+        self.__iter = 0
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.GET_NEW_EXP)
+        self.__chan_str_reqs.send_data(request)
+        self.__wait_exp_reqst = True
+
+    def __validate_exp_ranges(self, battery: CyclerDataBatteryC,
+                       profile: CyclerDataProfileC) -> bool:
+        """Returns True if the experiment is valid, which means the electric range of the profile
+        is between the limits of the battery.
+        
+        If the profile has no current neither voltage, means the profile could be only of waits, so
+        is also valid.
+        If the profile has no current or voltage, means the profile could have only current/voltage
+        modes apart from waits, so is also valid if the range of the profile is inside the range of
+        battery
+
+        Args:
+            battery (CyclerDataBatteryC): [description]
+            profile (CyclerDataProfileC): [description]
+
+        Returns:
+            bool: [description]
+        """
+        # Check if the profile range is inside the battery range
+        res = False
+        if profile.range.no_voltage() and profile.range.no_current():
+            log.debug("Profile no current neither voltages")
+            res = True
+        elif profile.range.no_current():
+            log.debug("Profile no current")
+            res = profile.range.in_range_voltage(battery.elec_ranges)
+        elif profile.range.no_voltage():
+            log.debug("Profile no voltage")
+            res = profile.range.in_range_current(battery.elec_ranges)
+        else:
+            log.debug("Profile with current and voltage")
+            res = (profile.range.in_range_current(battery.elec_ranges) and
+                profile.range.in_range_voltage(battery.elec_ranges))
+        log.critical(f"Validating experiment: {res}")
+        return res
+
+    def __update_exp_status(self, exp_status: CyclerDataExpStatusE) -> None:
+        """Write the experiment status in the shared memory
+
+        Args:
+            exp_status (CyclerDataExpStatusE): Experiment status
+        """
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.SET_EXP_STATUS,
+                        exp_status= exp_status)
+        self.__chan_str_reqs.send_data(request)
```

Comment:
You should set a flag that you are waiting for an experiment status response?

Reply:
Future versions